api/src/pcapi/core/providers/allocine_movie_list.py

```diff
@@ -0,0 +1,60 @@
+from sqlalchemy import Integer
+
+from pcapi.connectors.serialization.allocine_serializers import AllocineMovie
+from pcapi.core.categories.subcategories_v2 import SEANCE_CINE
+from pcapi.core.offers.models import OfferExtraData
+from pcapi.core.offers.models import Product
+from pcapi.domain import allocine as allocine_domain
+from pcapi.models import db
+from pcapi.repository import transaction
+
+
+def synchronize_products() -> None:
+    movies = allocine_domain.get_movie_list()
+    allocine_ids = [movie.internalId for movie in movies]
+    products_query = Product.query.filter(Product.extraData["allocineId"].cast(Integer).in_(allocine_ids))
+    products = products_query.all()
+    products_by_allocine_id = {product.extraData["allocineId"]: product for product in products}
```

Comment:
tu peux faire ça directement non?
`products_by_allocine_id = {product.extraData["allocineId"]: product for product in products_query}`

Reply:
oui mais est ce que ça n'utilise pas plus de memoire comme avec le .all?

Reply:
`list(query)` et `query.all()` sont totalement équivalents, j'espère que ça répond à ta question

Reply:
pas vraiment. ce n'ai pas ce que je compare on peux rediscuter ça en live une autre fois